__tests__/users/user-management-home-screen.test.js

```diff
@@ -0,0 +1,180 @@
+      if (url === 'https://api.realdevsquad.com/users?size=100&page=0') {
+        interceptedRequest.respond({
+          status: 200,
+          contentType: 'application/json',
+          headers: {
+            'Access-Control-Allow-Origin': '*',
+            'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE, OPTIONS',
+            'Access-Control-Allow-Headers': 'Content-Type, Authorization',
+          },
+          body: JSON.stringify(allUsersData),
+        });
+      } else if (url === 'https://api.realdevsquad.com/users?search=randhir') {
+        interceptedRequest.respond({
+          status: 200,
+          contentType: 'application/json',
+          headers: {
+            'Access-Control-Allow-Origin': '*',
+            'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE, OPTIONS',
+            'Access-Control-Allow-Headers': 'Content-Type, Authorization',
+          },
+          body: JSON.stringify(filteredUsersData),
+        });
+      } else {
+        interceptedRequest.continue();
+      }
+    });
+    await page.goto('http://localhost:8000/users');
+    await page.waitForNetworkIdle();
+
+    userListElement = await page.$('#user-list');
+    tileViewBtn = await page.$('#tile-view-btn');
+    tableViewBtn = await page.$('#table-view-btn');
+    userSearchElement = await page.$('#user-search');
+    paginationElement = await page.$('#pagination');
+    prevBtn = await page.$('#prevButton');
+    nextBtn = await page.$('#nextButton');
+  });
+
+  afterAll(async () => {
+    await browser.close();
+  });
+
+  it('Checks the UI elements on user listing page.', async () => {
+    expect(userListElement).toBeTruthy();
+    expect(tileViewBtn).toBeTruthy();
+    expect(tableViewBtn).toBeTruthy();
+    expect(userSearchElement).toBeTruthy();
+    expect(paginationElement).toBeTruthy();
+    expect(prevBtn).toBeTruthy();
+    expect(nextBtn).toBeTruthy();
+  });
+
+  it('Check the UI interactions of tile view and table view button.', async () => {
+    await tileViewBtn.click();
+    const activeTileView = await page.$('.btn-active');
+    expect(activeTileView).toBeTruthy();
+    expect(activeTileView).toEqual(tileViewBtn);
+
+    await tableViewBtn.click();
+    const activeTableView = await page.$('.btn-active');
+    expect(activeTableView).toBeTruthy();
+    expect(activeTableView).toEqual(tableViewBtn);
+  });
+
+  it('Check the home screen contain user-cards', async () => {
+    const userList = await page.$('#user-list');
+    const liList = await userList.$$('li');
+    expect(liList.length).toBeGreaterThan(0);
+  });
+
+  it('checks the search functionality to display queried user', async () => {
+    await page.type('input[id="user-search"]', 'randhir');
+    await page.waitForNetworkIdle();
+    await page.waitForFunction(() => {
+      const userList = document.querySelector('#user-list');
+      const userCard = userList.querySelectorAll('li');
+      return userCard.length === 1;
```

Comment:
in the actual case it will list all the users whose name begin with `randhir`. In this case I have intercepted the request and sent just 1 user so checked for length 1.

Reply:
If we don't check then wont our tests deviate from real world scenario?

Reply:
to check for real world scenario. I have modified the condition now.